src/include/main/attached_database.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <string>
+
+#include "function/table_functions.h"
+
+namespace kuzu {
+namespace main {
+
+class AttachedDatabase {
+
+public:
+    AttachedDatabase(std::string dbName, function::TableFunction scanFunction)
+        : dbName{std::move(dbName)}, scanFunction{std::move(scanFunction)} {}
+
+    std::string getDBName() { return dbName; }
+
+    function::TableFunction* getScanFunction() { return &scanFunction; }
```

Comment:
I wonder if it is better to just copy TableFunction?

Reply:
We can avoid an additional copy if we return by pointer and the funciton implementation is always the same.